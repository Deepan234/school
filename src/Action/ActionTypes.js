export const ActionTypes = {
    GET_SCHOOLS :"GET_SCHOOLS",
    INDUIVAL_SCHOOLS : "INDUIVAL_SCHOOLS",
    LOGIN : "LOGIN",
    UPDATE_SCHOOL:"UPDATE_SCHOOL",
    DELETE_SCHOOL:"DELETE_SCHOOL",
    ADD_SCHOOL:"ADD_SCHOOL"
}