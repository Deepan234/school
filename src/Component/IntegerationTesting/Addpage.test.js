import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import AddSchoolPage from '../Pages/AddSchoolPage';
import { addSchoolData } from '../../Action/SchoolAction';
const mockStore = configureStore([]);

describe ('My Connected React_redux Component ',()=>{
    let store;
    let component;
    beforeEach(()=>{
     store = mockStore(
         {
             name:'deepan school',
             about:'good'
         }
     )

     component = renderer.create(
         <Provider store={store}>
            <AddSchoolPage/>
         </Provider>
     )
    });

    test('should render from test store',()=>{
        expect(component.toJSON()).toMatchSnapshot();
    })

});
