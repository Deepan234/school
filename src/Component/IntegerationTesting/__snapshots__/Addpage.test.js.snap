// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React_redux Component  should render from test store 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <form>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="name"
        >
          NAME
        </label>
        <input
          className="form-control"
          id="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="about"
        >
          ABOUT
        </label>
        <input
          className="form-control"
          id="about"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="location"
        >
          LOCATION
        </label>
        <input
          className="form-control"
          id="location"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="contactNo"
        >
          CONTACT NO
        </label>
        <input
          className="form-control"
          id="contactNo"
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="distance"
        >
          DISTANCE
        </label>
        <input
          className="form-control"
          id="distance"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="emailId"
        >
          EMAIL ID
        </label>
        <input
          className="form-control"
          id="emailId"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="free"
        >
          FEE
        </label>
        <input
          className="form-control"
          id="free"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="affilation"
        >
          AFFILATION
        </label>
        <input
          className="form-control"
          id="affilation"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="rating"
        >
          RATING
        </label>
        <input
          className="form-control"
          id="rating"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="website"
        >
          WEBSITE
        </label>
        <input
          className="form-control"
          id="website"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="text-center"
      >
        <button
          className="btn btn-primary  w-25"
          id="submit-btn"
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
